R/lsmeans.R

```diff
@@ -2,37 +2,73 @@
 #' Least Square Means
 #'
 #'
-#' Estimates the least square means from a linear model. This is done by
-#' generating a prediction from the model using an hypothetical observation
-#' that is constructed by averaging the data. See details for more information.
+#' Estimates the least square means from a linear model. The exact implementation
+#' / interpretation depends on the weighting scheme; see the weighting section for more
+#' information.
 #'
-#' @details
+#' 
+#' @section Weighting:
 #'
-#' The lsmeans are obtained by calculating hypothetical patients
-#' and predicting their expected values. These hypothetical patients
-#' are constructed by expanding out all possible combinations of each
-#' categorical covariate and by setting any numerical covariates equal
-#' to the mean.
+#' ### Counterfactual
 #'
-#' A final lsmean value is calculated by averaging these hypothetical
-#' patients. If `.weights` equals `"proportional"` then the values are weighted
-#' by the frequency in which they occur in the full dataset. If `.weights`
-#' equals `"equal"` then each hypothetical patient is given an equal weight
-#' regardless of what actually occurs in the dataset.
+#' For `weights = "counterfactual"` (the default) the lsmeans are obtained by
+#' taking the average of the predicted values for each patient after assigning all patients
+#' to each arm in turn.
+#' This approach is equivalent to standardization or g-computation.
+#' In comparison to `emmeans` this approach is equivalent to:
+#' ```
+#' emmeans::emmeans(model, specs = "<treatment>", counterfactual = "<treatment>")
+#' ```
+#' Note that `weights = "proportional"` is an alias for `weights = "counterfactual"`.
+#' Whilst potentially confusing, this has been done to ensure backwards compatibility
```

Comment:
Do we really need to mention this in the documentation? Maybe something like "This is not equivalent to `emmeans`'s `weights = "proportional"`. To get results consistent with the `"proportional"` weighting scheme of `emmeans`, please use `weights = "proportional_em"`."?

Reply:
ok!